pkg/handlers/generic/mutation/imageregistries/credentials/inject.go

```diff
@@ -359,3 +374,27 @@ func secretForImageRegistryCredentials(
 	err := c.Get(ctx, key, secret)
 	return secret, err
 }
+
+// needImageRegistryCredentials will return true if all the providerConfigs have valid credentials
+// will return false if there is only a single providerConfig for a mirror with no credentials
+// otherwise will return an error.
```

Comment:
Honestly, I don't understand what this function is doing or why it is needed  😅 Let's chat about it later.

Reply:
Added better comments, lmk if that makes more sense 🙏